src/libs/ReportUtils.js

```diff
@@ -1874,44 +1897,54 @@ function isUnreadWithMention(report) {
 }
 
 /**
- * Determines if a report has an outstanding IOU that doesn't belong to the currently logged in user
+ * Determines if a report has an outstanding IOU that is waiting for an action from the current user
  *
  * @param {Object} report
  * @param {String} report.iouReportID
- * @param {Object} iouReports
  * @returns {boolean}
  */
-function hasOutstandingIOU(report, iouReports) {
-    if (!report || !report.iouReportID || _.isUndefined(report.hasOutstandingIOU)) {
-        return false;
+function isWaitingForIOUActionFromCurrentUser(report) {
+    let reportToLook = report;
+
+    if (report.iouReportID) {
```

Comment:
I'm not calling this method anymore for the green dot logic. Just for the logic of whether we need to display the report on the left panel or not.

But yes, the idea is:
- If it's the chat report, check if its child iou report has an action to do for the current user
- If it's the iou report, check  if it has an action to do for the current user

This way, we're sure both are displayed on the left.
Green dot does show on both reports (chat and iou) for the payer when they need to pay (because they have a button to pay on either).
But yes, for the waitingForBankAccount case, the green dot only shows on the iou report.